pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -675,12 +675,6 @@ func (oe *operationExecutor) MountVolume(
 	}
 
 	podName := nestedpendingoperations.EmptyUniquePodName
-	// TODO: remove this -- not necessary
-	if !volumeToMount.PluginIsAttachable {
-		// Non-attachable volume plugins can execute mount for multiple pods
```

Comment:
I think we do want to allow non-attachable volume execute mount concurrently. Not sure why there is TODO: remove this.
cc @saad-ali